Tests/URLRoutingTests/URLRoutingTests.swift

```diff
@@ -97,6 +97,54 @@ class URLRoutingTests: XCTestCase {
       URLRequestData(query: [:])
     )
   }
+  
+  func testFragment() throws {
+    // test parser builder
+    let p1 = Fragment {
+      Rest()
+    }
+    
+    var request = URLRequestData(fragment: "fragment")
+    let fragment = try p1.parse(&request)
+    XCTAssertEqual("fragment", fragment)
+    XCTAssertEqual(request.fragment, "")
+    
+    XCTAssertEqual(
+      try p1.print("fragment"),
+      .init(fragment: "fragment")
+    )
+    
+    // a toy conversion to uppercase for testing purposes
+    struct UppercaseConversion: Conversion {
+      func apply(_ input: Substring) throws -> Substring {
+        String(input).uppercased()[...]
+      }
+      func unapply(_ output: Substring) throws -> Substring {
+        String(output).lowercased()[...]
+      }
+    }
+    
+    // test conversion
+    let p2 = Fragment(UppercaseConversion())
+    
+    XCTAssertEqual(
+      "FRAGMENT",
+      try p2.parse(.init(fragment: "fragment"))
+    )
+    XCTAssertEqual(
+      .init(fragment: "fragment"),
+      try p2.print("FRAGMENT")
+    )
+    
+    // test conversion with parser builder
+    let p3 = Fragment(UppercaseConversion()) {
+      Prefix(4)
+    }
+    
+    request = .init(fragment: "frag")
+    XCTAssertEqual("FRAG", try p3.parse(&request))
+    XCTAssertEqual(.init(fragment: "frag"), try p3.print("FRAG"))
+  }
```

Comment:
I'm missing the case when a fragment contains a value - like `#id=value`. Maybe the [`testBaseURL`](https://github.com/pointfreeco/swift-url-routing/blob/9795f8fac9266af084f9c8c745265ce222b2c0fd/Tests/URLRoutingTests/URLRoutingTests.swift#L192) test should be extended too by a fragment component. And another candidate for extension is maybe the [routing benchmark](https://github.com/pointfreeco/swift-url-routing/blob/d68e1695de7244f69919f718a36b532260475f40/Sources/swift-url-routing-benchmark/Routing.swift).